### PR TITLE
[🐴] Rich text in messages

### PR DESCRIPTION
**Pull request by mozzius**
Adds support for links, mentions, tags. Not yet displayed while typing, it generates them just as you send and then the bubble can now display them

Screenshot 2024-05-09 at 13 42 07

### CODE REVIEW

File: src/components/RichText.tsx [hunk 1]
```diff
@@ -35,13 +36,22 @@ export function RichText({
...
-  const styles = [a.leading_snug, flatten(style)]
+
+  const flattenedStyle = flatten(style)
+  const plainStyles = [a.leading_snug, flattenedStyle]
+  const interactiveStyles = [
+    a.leading_snug,
+    a.pointer_events_auto,
+    display === 'underlines' && a.underline,
+    flattenedStyle,
+  ]
```
Comment (estrattonbailey) [inline src/components/RichText.tsx:54]: Wait why is this necessary here 🤔

Comment (mozzius) [reply src/components/RichText.tsx:54]: It was an attempt to simplify the code here, there was a lot of repetition of `[...styles, {pointerEvents: 'auto'}]` and I wanted to add `a.underline` to it

Comment (mozzius) [reply src/components/RichText.tsx:54]: I can't just pass in underline to the style prop because that would affect all the text

Comment (estrattonbailey) [reply src/components/RichText.tsx:54]: Oh of course, I got ya, my b. Kind of a nit, but we have `style`, can we just do `interactivesStyle` or something and pass in styles as an array that get spread here in place of the `display` check? Feels more stable than these other values.

Comment (mozzius) [reply src/components/RichText.tsx:54]: I like that better, will do
